### DISCUSSION TITLE
dependecy control system

### DISCUSSION THREAD

**Discussion by 1337180** [joomla/joomla-cms#32383] [2. Cold Features]
Will have J4 dependency control manageable system API for plugin developers with standardized/alternative versions?
Like we want to developers IF then use latest officially supported versions. Like VUE3 vuex4 and so.

**Comment by dgrammatiko** [comment]
You **don't** need some Joomla specific system. All modern js scripts are module-based and work with imports/exports eg https://github.com/joomla/joomla-cms/blob/38dd8f04ccb63095accc74b15d0343ffb5975848/administrator/components/com_media/resources/scripts/mediamanager.es6.js#L1-L23 . So in the current 4.0 branch, the media manager has a dependency on the vue 2.6.12 but if you write another front end app for, let's say installer you can have there vue 3.0.0. Joomla **can't** and shouldn't interfere on the front end (basically npm based) dependencies. Probably you're thinking of legacy scripts like jQuery et al but that's just legacy and you should treat them as legacy as well.

My 2p
